examples/bin/run-druid

```diff
@@ -42,7 +42,7 @@ if [ -z "$JAVA_BIN" ]; then
   exit 1
 fi
 
-LOG_DIR="${DRUID_LOG_DIR:=${WHEREAMI}/log}"
+LOG_DIR="${DRUID_LOG_DIR:=${WHEREAMI}/../log}"
```

Comment:
https://github.com/apache/druid/blame/master/docs/operations/single-server.md#L51

The doc launches the server at druid directory, so this change will put log at this directory: 
druid_directory/../log

I think it's more intuitive to create the log directory at the directory where user executes the startup script.

Reply:
> I think it's more intuitive to create the log directory at the directory where user executes the startup script.

That's what I thought I was doing! 🙂

When I ran the `bin/start-micro-quickstart` command like we have in the tutorials, with the script currently in master, logs were created in `bin/log/` in the root of the distribution. This seemed strange so I wanted to change it to `log/`. So that's why I added the `../`.

Reply:
@FrankChen021 just wanted to double check with you to see if I am missing anything here. I think the `../` ends up doing more what you intended in https://github.com/apache/druid/pull/10147 anyway.

(Since `$WHEREAMI` is the `bin` directory, `$WHEREAMI/..` is the Druid distribution directory.)

Reply:
The location depends on the directory you are in when you launch Druid. Sounds like @gian launches Druid from within the `$DRUID_HOME/bin` directory. But, I always do it from `$DRUID_HOME`: `./bin/start-micro-quickstart` which is how Apache tools usually work.

A better place to put Druid logs is `$DRUID_HOME/var/druid`, which is where `derby.log`, `indexing-logs` and others end up. If that directory is a bit too general, then perhaps `$DRUID_HOME/var/log/druid` (moving the other logs to match).

It appears that `$WHEREAMI` in `start-micro-quickstart` is the directory that contains the script, so `$DRUID_HOME/bin`. The same is true in `bin/run-druid`. It is odd, however, that `start-micro-quickstart` does not call `run-druid`, so any changes you make in `run-druid` won't affect `start-micro-quickstart`.

I would argue this line in `run-druid` is wrong:

```bash
LOG_DIR="${DRUID_LOG_DIR:=${WHEREAMI}/log}"
```

That will put the log in `$DRUID_HOME/bin/log` which seems quite odd. Maybe change it to:

```bash
LOG_DIR="${DRUID_LOG_DIR:=${WHEREAMI}/../var/log/druid}"
```

Reply:
I remember that 

- if `bin/start-micro-quickstart` is executed at `$DRUID_HOME`, the log is generated at `$DRUID_HOME/log`.
- If the script is started at the `bin` directory, that is `$DRUID_HOME/bin`, the log is definitely generated at `$DRUID_HOME/bin/log`. This dir is what you think not a good place to save the logs.

Let me check again.

Reply:
@FrankChen021 - did you get a chance to check it?

Reply:
I just tried it myself and the log is indeed getting generated at `$DRUID_HOME/bin/log` even if I run the script outside `bin` folder. I am going to merge this and get it into `0.23.0` branch to make the RC build. 
@FrankChen021 - if you find anything different in your testing, please do let us know. If any changes need to be made, we can do it in a follow-up PR.

Reply:
Sorry for the late reply due to busy work.

I started the `bin/start-micro-quickstart` at the druid home directory, it turned out that log files are generated at `bin\log` which is not my intention. I remember that they should be under `$DRUID_HOME/log`, but they're not. Maybe my memory is wrong.

Based on current behaviour, this PR fixes the problem. Thank you @gianm for fixing this.